### PR TITLE
wallet cli: Remove manually created help text, use clap's text instead.

### PR DESCRIPTION
**Pull request by mvines**

### CODE REVIEW

File: src/bin/wallet.rs [hunk 1]
```diff
@@ -155,7 +155,7 @@ fn parse_args() -> Result<WalletConfig, Box<error::Error>> {
         path.to_str().unwrap()
     };
     let id = read_keypair(id_path).or_else(|err| {
-        display_actions();
+        println!("{}", matches.usage());
```
Comment (garious) [inline src/bin/wallet.rs]: Should these all be `eprintln!`?

Comment (mvines) [reply src/bin/wallet.rs]: I don't have much of an opinion right now, we aren't super consistent about this.  The previous help went to stdout and this patch preserves that behavior.  Happy to add an `e`, lmk

**Comment by mvines** [comment]
The CI failure here is the flaky `test_multi_node_dynamic_network` test :-/

File: src/bin/wallet.rs (hunk 1 above)
Comment (garious) [reply src/bin/wallet.rs]: Yeah, I think traditionally you'd only write `usage()` to stdout if `--help`.

File: src/bin/wallet.rs [hunk 2]
```diff
@@ -177,7 +177,7 @@ fn parse_args() -> Result<WalletConfig, Box<error::Error>> {
                     .expect("base58-encoded public key");
 
                 if pubkey_vec.len() != std::mem::size_of::<PublicKey>() {
-                    display_actions();
+                    println!("{}", matches.usage());
```
Comment (CriesofCarrots) [inline src/bin/wallet.rs]: This is a bit nicer if you use `pay_matches.usage()`

File: src/bin/wallet.rs [hunk 3]
```diff
@@ -198,14 +198,14 @@ fn parse_args() -> Result<WalletConfig, Box<error::Error>> {
                 let sig = Signature::new(&sig_vec);
                 Ok(WalletCommand::Confirm(sig))
             } else {
-                display_actions();
+                println!("{}", matches.usage());
```
Comment (CriesofCarrots) [inline src/bin/wallet.rs]: Similarly, I suggest: `confirm_matches.usage()`

File: src/bin/wallet.rs (hunk 1 above)
Comment (CriesofCarrots) [inline src/bin/wallet.rs]: I'm not sure that this print statement actually provides anything of value here (about avoiding a keypair error). Maybe just remove it entirely?

File: src/bin/wallet.rs [hunk 4]
```diff
@@ -198,14 +198,14 @@ fn parse_args() -> Result<WalletConfig, Box<error::Error>> {
...
                 Err(WalletError::BadParameter("Invalid signature".to_string()))
             }
         }
         ("balance", Some(_balance_matches)) => Ok(WalletCommand::Balance),
         ("address", Some(_address_matches)) => Ok(WalletCommand::Address),
         ("", None) => {
-            display_actions();
+            println!("{}", matches.usage());
```
Comment (CriesofCarrots) [inline src/bin/wallet.rs]: Hmm. I am dissatisfied with `matches.usage()` here, as it doesn't print the possible subcommands. I don't see an immediate way to get clap to print out the help text instead, but that would be my preference. I'll look into it briefly. That said, there are plenty of CLIs out there that do it this way, so not a big deal. I just think it's not the nicest UX.

Comment (mvines) [reply src/bin/wallet.rs]: yeah I guess you add a `-h` or `--help` to the command-line when you arrive at this message and need more help :)

Comment (CriesofCarrots) [reply src/bin/wallet.rs]: Yep, just an extra step.

**Comment by mvines** [comment]
PR updated with review feedback addressed

**Pull request merged**